### PR TITLE
Add types to wait_for_message.py and moves Handles into type stubs

### PR DESCRIPTION
**Pull request by InvincibleRMC**
This pull request adds types to wait_for_message.py. It also creates the `_rclpy.WaitSet`.  Finally it moves the various `Handle` objects into _rclpy_pybind11.pyi.

### CODE REVIEW

File: rclpy/rclpy/wait_for_message.py [hunk 1]
```diff
@@ -41,6 +42,10 @@ def wait_for_message(
         could not be obtained or shutdown was triggered asynchronously on the context.
     """
     context = node.context
+
+    if context.handle is None:
+        raise Exception('Cannot create Waitset without a context.handle')
```
Comment (sloretz) [inline rclpy/rclpy/wait_for_message.py:47]: Nit, please use `RuntimeError` here.

**Review by sloretz — CHANGES_REQUESTED** [review]
LGTM with on nitpick addressed

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
Pulls: ros2/rclpy#1325
Gist: https://gist.githubusercontent.com/sloretz/e11b408ea5abaf83d4be1cab5bdf4568/raw/e737363e25007355832cccf984cd87285e962c19/ros2.repos
BUILD args:  --packages-above-and-dependencies rclpy
TEST args:  --packages-above rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14436
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21704/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15978/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1209/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22408/)

**Comment by sloretz** [comment]
@mergifyio update

**Comment by sloretz** [comment]
Pulls: ros2/rclpy#1325
Gist: https://gist.githubusercontent.com/sloretz/62f40dc912a8c6d90383b8b429f90d84/raw/e737363e25007355832cccf984cd87285e962c19/ros2.repos
BUILD args:  --packages-above-and-dependencies rclpy
TEST args:  --packages-above rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14437
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21705/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15979/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1210/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22409/)

**Comment by InvincibleRMC** [comment]
There seems to be a merge problem I will fix when I get a chance.

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
@mergifyio update

**Comment by sloretz** [comment]
Pulls: ros2/rclpy#1325
Gist: https://gist.githubusercontent.com/sloretz/f76824db58100e24ade28109e5780bfb/raw/e737363e25007355832cccf984cd87285e962c19/ros2.repos
BUILD args:  --packages-above-and-dependencies rclpy
TEST args:  --packages-above rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14456
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21725/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16000/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1230/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22428/)

**Pull request merged**